src/components/UserProfile/UserProfileContainer.tsx

```diff
@@ -0,0 +1,96 @@
+import { Accordion, AccordionDetails, AccordionSummary, Typography } from "@mui/material";
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import UserProfile from "./UserProfile";
+import { useDispatch, useSelector } from "react-redux";
+import { isLoaded, loading, profileState, setProfile } from "../../state-slices/user-profile/profile-slice";
+import { useEffect } from "react";
+import { getUserData } from "../../remote/user-service";
+import { UserData } from "../../models/user-data";
+import { User } from "../../models/user";
+import { authState } from "../../state-slices/auth/auth-slice";
+
+const UserProfileContainer = (props: any) => {
+  const state = useSelector(profileState);
+  const dispatch = useDispatch();
+  const user: User = useSelector(authState).authUser;
+
+
+  const getData = async function(){
+      try{
+          dispatch(loading());
```

Comment:
What does this do?

Reply:
Since we have to wait for the lambdas to spin up, the user's information won't be retrieved quickly enough. That causes an undefined exception. There's a loading boolean inside of the profile slice and loading/isloaded reducers (and every other slice for that matter) to set a piece of state to be used as in ternary to determine whether or not the information is ready to be displayed.